Update doc for metric cardinality and disabled metric (#27069)
kubernetes/website: content/en/docs/concepts/cluster-administration/system-metrics.md

The metrics are exposed at the HTTP endpoint `/metrics/resources` and require the same authorization as the `/metrics`
endpoint on the scheduler. You must use the `--show-hidden-metrics-for-version=1.20` flag to expose these alpha stability metrics.

> ## Disable problematic metrics
> 
> You can explicitly turn off misbehaving metrics via CLI flag `--disabled-metrics`. The input is a list of disabled metrics(i.e. `--disabled-metrics=metric1,metric2`).

Reviewer: ```suggestion
## Disabling metrics

You can explicitly turn off metrics via CLI flag `--disabled-metrics`. This may be desired if, for example, a metric is causing a performance problem. The input is a list of disabled metrics (i.e. `--disabled-metrics=metric1,metric2`).
```

Author: updated as suggested. Thanks!

> You can explicitly turn off metrics via CLI flag `--disabled-metrics`. This may be desired if, for example, a metric is causing a performance problem. The input is a list of disabled metrics (i.e. `--disabled-metrics=metric1,metric2`).

Reviewer 2: “CLI” typically implies `kubectl`. I recommend rewording to make it clear where you'd put that command line argument.

Author: Done. Thanks

> ## Metric Cardinality Enforcement

Reviewer 2: style nit:
```suggestion
## Metric cardinality enforcement
```

Author: Updated

> Metrics with unbounded dimensions could cause memory issues in the components they instrument. To solve this problem, we introduce a flag `--allow-label-value` to  dynamically configure an allow-list of label values for a metric.

Reviewer 2: Terms like “introduce” are good for blogs and release notes; in concept docs we assume the reader is thinking about the feature in-use, so we use different wording such as
```suggestion
Metrics with unbounded dimensions could cause memory issues in the components they
instrument. To limit resource use, you can use the `--allow-label-value` command line
option to dynamically configure an allow-list of label values for a metric.
```

Author: Got you. changed.

> In alpha stage, the flag can only take in the allow-list value which is a [StringToString](https://pkg.go.dev/github.com/spf13/pflag#StringToStringVar) type argument from the CLI with the format 

Reviewer 2: Avoid assuming that the reader can write Golang (many cluster operators don't / can't). For example:
```suggestion
This command line option takes a series of mappings. Each mapping is a of the format
`<metric_name>,<allowed_labels>` where the `<allowed_labels>` part is a comma-separated
list of acceptable label names.

The overall format looks like:
```

Author: Reworded.

`--allow-label-value <metric_name>,<label_name>='<allow_value1>, <allow_value2>...', <metric_name2>,<label_name>='<allow_value1>, <allow_value2>...', ...`.

> Here is an example input.

Reviewer 2: ```suggestion
Here is an example:
```

I would use realistic-looking names rather than “metric1`, “label2”, etc.

Author: Changed to some meaningful names.

The metrics are exposed at the HTTP endpoint `/metrics/resources` and require the same authorization as the `/metrics`
endpoint on the scheduler. You must use the `--show-hidden-metrics-for-version=1.20` flag to expose these alpha stability metrics.

## Disabling metrics

You can explicitly turn off metrics via command line flag `--disabled-metrics`. This may be desired if, for example, a metric is causing a performance problem. The input is a list of disabled metrics (i.e. `--disabled-metrics=metric1,metric2`).

## Metric cardinality enforcement

Metrics with unbounded dimensions could cause memory issues in the components they instrument. To limit resource use, you can use the `--allow-label-value` command line option to dynamically configure an allow-list of label values for a metric.

> In alpha stage, the flag can only take in a series of mappings as metric label allow-list. Each mapping is a of the format `<metric_name>,<label_name>=<allowed_labels>` where the `<allowed_labels>` part is a comma-separated list of acceptable label names.

Reviewer 3: ```suggestion
In alpha stage, the flag can only take in a series of mappings as metric label allow-list.
Each mapping is of the format `<metric_name>,<label_name>=<allowed_labels>` where
`<allowed_labels>` is a comma-separated list of acceptable label names.
```

Author: Fixed. Thanks for finding this typo.
